be/src/exec/pipeline/exchange/local_exchange_memory_manager.h

```diff
@@ -23,15 +23,17 @@ namespace starrocks::pipeline {
 // Manage the memory usage for local exchange
 class LocalExchangeMemoryManager {
 public:
-    LocalExchangeMemoryManager(size_t max_input_dop) {
+    LocalExchangeMemoryManager(size_t max_input_dop) : _max_input_dop(max_input_dop) {
         if (config::local_exchange_buffer_mem_limit_per_driver > 0) {
             _max_memory_usage_per_driver = config::local_exchange_buffer_mem_limit_per_driver;
         } else {
             LOG(WARNING) << "invalid config::local_exchange_buffer_mem_limit_per_driver "
                          << config::local_exchange_buffer_mem_limit_per_driver;
         }
         size_t res = max_input_dop * _max_memory_usage_per_driver;
-        _max_memory_usage = (res > _max_memory_usage || res <= 0) ? _max_memory_usage : res;
+        if (res < _max_memory_usage) {
```

Comment:
res != 0?

Reply:
> res != 0?

res won't be 0. if config:: local_exchange_buffer_mem_limit_per_driver < 0, it will use 128M as default value and max_input_dop must be greater than 0, so res will always greater than 0.